src/Workspaces/Core/Portable/Workspace/Host/EventListener/WellKnownEventListeners.cs

```diff
@@ -11,6 +11,5 @@ internal static class WellKnownEventListeners
     {
         public const string Workspace = nameof(Workspace);
         public const string DiagnosticService = nameof(DiagnosticService);
-        public const string TaskListProvider = nameof(TaskListProvider);
```

Comment:
one step closer to removing all this craziness.

Reply:
No objection to "removing this craziness", but I'd like to see it done _without_ coupling the result to `VisualStudioWorkspace`.